### PR TITLE
nfd-master: fix --label-whitelist

### PR DESCRIPTION
**Pull request by marquiz**
Make the --label-whitelist effective. Previously, it was unused and had
no effect. Also, add simple unit test for that.

### CODE REVIEW

File: pkg/nfd-master/nfd-master.go [hunk 1]
```diff
@@ -290,13 +290,14 @@ func updateNodeFeatures(helper apihelper.APIHelpers, nodeName string, labels Lab
 	removeLabelsWithPrefix(node, "node.alpha.kubernetes-incubator.io/node-feature-discovery")
 
 	// Add labels to the node object.
+	filterFeatureLabels(&labels, s.args.LabelWhiteList)
```
Comment (Ethyling) [inline pkg/nfd-master/nfd-master.go]: I think we must filter the labels before the annotation is generated no? Here we are adding all the labels in the annotation and then filter the labels using the regex.

Comment (marquiz) [reply pkg/nfd-master/nfd-master.go]: Yup, you're correct 🤭 I obviously wrote this patch a bit too hastily.

This is now fixed and the unit test is updated to check this. I also added another patch fixing a former issue in unit tests

**Comment by marquiz** [comment]
I pushed a new version where whitelist filter is applied even if --no-publish is specified

**Comment by Ethyling** [comment]
It seems like the regexp in the function `filterFeatureLabels` is `NULL` when we are running the tests `TestRun` and `TestRunTls` from `pkg/nfd-worker/nfd-worker_test.go`. I don't know if the fix is to pass a regexp in the master args?

**Comment by marquiz** [comment]
*Arrgh*, I hope I got it right, this time 😊

**Pull request merged**